### PR TITLE
Proposal for consent section

### LINKED ISSUE #9: Old: Terminology -- whether to use path signals or something else
Ted is concerned that terms may cause past fights to flare up again

### PR DESCRIPTION
**Pull request by mirjak**
closes #9

### CODE REVIEW

**Review by jariarkko — APPROVED** [review]
Looks pretty good to me! Thanks.

File: draft-iab-path-signals-collaboration.md [hunk 1]
```diff
@@ -301,41 +301,43 @@ an airgapped network.  Most "closed" networks have at least some needs
...
-
-  As a result, typically an application's consent is not the same as
-  the user's consent.
+## Control of the Distribution of Information
+
+Trust and mutual agreement between the involved entities must determine
+the distribution of information, in order to give adequate control to 
+each entity over the collaboration or information sharing.
+
+A corollary of the intentional distribution is that the sender
```
Comment (tfpauly) [inline draft-iab-path-signals-collaboration.md:310]:
```suggestion
A corollary of this principle is that the sender
```

File: draft-iab-path-signals-collaboration.md [hunk 2]
```diff
@@ -301,41 +301,43 @@ an airgapped network.  Most "closed" networks have at least some needs
...
+needs to agree to sending the information. Or that the requester
+for an action needs to agree to make a request; it should not be an
+implicit decision by the receiver that information was sent or a
+request was made, just because a packet happened to be formed in a
+particular way.
```
Comment (tfpauly) [inline draft-iab-path-signals-collaboration.md:315]: The grammar here seems odd, and I'm not quite sure what it's trying to say? Is this really a corollary?

**Review by tfpauly — CHANGES_REQUESTED** [review]

File: draft-iab-path-signals-collaboration.md (hunk 1 above)
Comment (jariarkko) [reply draft-iab-path-signals-collaboration.md:310]: Right. Thanks Tommy.

File: draft-iab-path-signals-collaboration.md (hunk 2 above)
Comment (mirjak) [reply draft-iab-path-signals-collaboration.md:315]: I didn't write this text originally. Is it any better if we just say "consequence" instead or "Following this principle"?

Comment (jariarkko) [reply draft-iab-path-signals-collaboration.md:315]:
```suggestion
The sender
needs to agree to sending the information. Any passing of information or request
for an action needs to be explicit, and use protocol mechanisms that are designed
for the purpose. Merely sending a particular kind of a packet to a destination should not
be interpreted as an implicit agreement.
```

Comment (jariarkko) [reply draft-iab-path-signals-collaboration.md:315]: In the above suggestion, we don't need to say whether this is a corollary or not. We can just describe the situation from the sender's perspective.

Comment (jariarkko) [reply draft-iab-path-signals-collaboration.md:315]: Please review.

File: draft-iab-path-signals-collaboration.md [hunk 3]
```diff
@@ -301,41 +301,43 @@ an airgapped network.  Most "closed" networks have at least some needs
...
+Trust and mutual agreement between the involved entities must determine
+the distribution of information, in order to give adequate control to 
+each entity over the collaboration or information sharing.
+
+A corollary of this principle is that the sender
+needs to agree to sending the information. Or that the requester
+for an action needs to agree to make a request; it should not be an
+implicit decision by the receiver that information was sent or a
+request was made, just because a packet happened to be formed in a
+particular way.
```
Comment (mirjak) [inline draft-iab-path-signals-collaboration.md:315]:
```suggestion
The sender needs to agree to sending the information.
Any passing of information or request for an action needs to be explicit,
and use protocol mechanisms that are designed for the purpose.
Merely sending a particular kind of packet to a destination should not
be interpreted as an implicit agreement.
```

File: draft-iab-path-signals-collaboration.md (hunk 2 above)
Comment (mirjak) [reply draft-iab-path-signals-collaboration.md]: I created a new commit with your suggested wording and merged it.

File: draft-iab-path-signals-collaboration.md [hunk 4]
```diff
@@ -301,41 +301,42 @@ an airgapped network.  Most "closed" networks have at least some needs
...
+In both cases, all involved entities must be identified and
+potentially authenticated if trust is required as a prerequisite
+to share certain information.
+
+Many Internet communications are not made for the applications, but
+ultimately made on behalf of users. However, no all information
+that may be shared does directly relate to user action or other
+senstive data. Still, all information shared must be evaluated carefully
+to identify potential privacy implications for users. Information that
+directly relates to the users should not be shared without the user's
```
Comment (mirjak) [inline draft-iab-path-signals-collaboration.md:333]:
```suggestion
directly relates to the user should not be shared without the user's
```

File: draft-iab-path-signals-collaboration.md [hunk 5]
```diff
@@ -301,41 +301,42 @@ an airgapped network.  Most "closed" networks have at least some needs
...
+"slow path" packet processing is required or implied, and the
+recipient or router did not have willingness for this.
+
+In both cases, all involved entities must be identified and
+potentially authenticated if trust is required as a prerequisite
+to share certain information.
+
+Many Internet communications are not made for the applications, but
+ultimately made on behalf of users. However, no all information
+that may be shared does directly relate to user action or other
```
Comment (mirjak) [inline draft-iab-path-signals-collaboration.md:330]:
```suggestion
that may be shared does directly relate to user actions or other
```

File: draft-iab-path-signals-collaboration.md (hunk 3 above)
Comment (jariarkko) [reply draft-iab-path-signals-collaboration.md]: Thanks.

File: draft-iab-path-signals-collaboration.md (hunk 4 above)
Comment (jariarkko) [reply draft-iab-path-signals-collaboration.md]: Ok

File: draft-iab-path-signals-collaboration.md [hunk 6]
```diff
@@ -306,40 +306,42 @@ an airgapped network.  Most "closed" networks have at least some needs
...
+each entity over the collaboration or information sharing.
+
+The sender needs to agree to sending the information.
+Any passing of information or request for an action needs to be explicit,
+and use protocol mechanisms that are designed for the purpose.
+Merely sending a particular kind of packet to a destination should not
+be interpreted as an implicit agreement.
+
+At the same time, the recipient of information or the target of a
+request should agree to wishing to receive the information. It
```
Comment (tfpauly) [inline draft-iab-path-signals-collaboration.md:322]:
```suggestion
request should agree to receiving the information. It
```

File: draft-iab-path-signals-collaboration.md [hunk 7]
```diff
@@ -306,40 +306,42 @@ an airgapped network.  Most "closed" networks have at least some needs
...
+willigness or a need to do so. This happens naturally in most
+protocol designs, but has been a problem for some cases where
+"slow path" packet processing is required or implied, and the
+recipient or router did not have willingness for this.
+
+In both cases, all involved entities must be identified and
+potentially authenticated if trust is required as a prerequisite
+to share certain information.
+
+Many Internet communications are not made for the applications, but
```
Comment (tfpauly) [inline draft-iab-path-signals-collaboration.md:333]:
```suggestion
Many Internet communications are not performed on behalf of the applications, but are
```

File: draft-iab-path-signals-collaboration.md [hunk 8]
```diff
@@ -306,40 +306,42 @@ an airgapped network.  Most "closed" networks have at least some needs
...
+ultimately made on behalf of users. However, no all information
```
Comment (tfpauly) [inline draft-iab-path-signals-collaboration.md:334]:
```suggestion
ultimately made on behalf of users. However, not all information
```

File: draft-iab-path-signals-collaboration.md [hunk 9]
```diff
@@ -306,40 +306,42 @@ an airgapped network.  Most "closed" networks have at least some needs
...
+that may be shared does directly relate to user actions or other
+senstive data. Still, all information shared must be evaluated carefully
```
Comment (tfpauly) [inline draft-iab-path-signals-collaboration.md:336]:
```suggestion
senstive data. All information shared must be evaluated carefully
```

**Review by tfpauly — APPROVED** [review]

**Pull request merged**